mods/page_save.js

```diff
@@ -120,12 +120,30 @@ PageSave.prototype._checkParams = function(params) {
     }
 };
 
+PageSave.prototype._checkRequestSource = function(req) {
+    var xff = req.headers['x-forwarded-for'];
+    var remoteAddr = xff && xff.split(',')[0].trim()
+        || req.socket && req.socket.remoteAddress;
+    if (!/^(?:::ffff:)?(?:10|127)\./.test(remoteAddr)) {
+        throw new rbUtil.HTTPError({
+            status: 403,
+            body: {
+                type: 'access_denied#revision',
+                title: 'Access to resource denied',
+                description: 'Only internal clients are allowed to use save api'
```

Comment:
IMHO, *This client is not allowed to use the save API* is better, as it still tells the client they can't do that, but doesn't expose the security reason behind it (which might incite them to try to get around it) :)